marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/triggers/mapping/cleanUpMapping.sjs

```diff
@@ -1,16 +1,19 @@
 'use strict';
-const es = require('/MarkLogic/entity-services/entity-services');
 declareUpdate();
 
 var uri;
 
 let xmlURI = fn.replace(uri, '\\.json$', '.xml');
 
-xdmp.invokeFunction(function() {
-    if (fn.docAvailable(xmlURI)) {
-      xdmp.documentDelete(xmlURI);
-      es.mappingDelete(xmlURI);
-    }
-  },
-  { update: "true", commit: "auto" }
-);
+xdmp.eval('if (fn.docAvailable("' + xmlURI + '")) { xdmp.documentDelete("' + xmlURI + '");xdmp.documentDelete("' + xmlURI  + '.xslt")}',
```

Comment:
This seems like it should have been breaking before, but there wasn't a test to catch it? I think it would be an easy test to write - have the suiteSetup insert a new mapping. The test can then verify that the .xslt file exists. Then the test can delete the mapping and then verify that the .xslt has been deleted (will need separate transactions in the test to achieve this).

Reply:
also please throw some kind of error and record if this fails. It's essential to know.

Reply:
@aebadirad ,
If it fails, it throws an exception, for example

13:16:51.437 [http-nio-8080-exec-7] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is com.marklogic.client.FailedRequestException: Local message: failed to apply resource at eval: Internal Server Error. Server Message: JS-JAVASCRIPT: if fn.docAvailable(/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml){xdmp.documentDelete("/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml");xdmp.documentDelete("/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml.xslt")} -- Error running JavaScript request: SyntaxError: Unexpected identifier] with root cause
com.marklogic.client.FailedRequestException: Local message: failed to apply resource at eval: Internal Server Error. Server Message: JS-JAVASCRIPT: if fn.docAvailable(/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml){xdmp.documentDelete("/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml");xdmp.documentDelete("/mappings/TestFlow-ccscs/TestFlow-ccscs-0.mapping.xml.xslt")} -- Error running JavaScript request: SyntaxError: Unexpected identifier
  at com.marklogic.client.impl.OkHttpServices.checkStatus(OkHttpServices.java:4395)
  at com.marklogic.client.impl.OkHttpServices.postIteratedResourceImpl(OkHttpServices.java:3917)
  at com.marklogic.client.impl.OkHttpServices.postEvalInvoke(OkHttpServices.java:3854)
  at com.marklogic.client.impl.ServerEvaluationCallImpl.eval(ServerEvaluationCallImpl.java:164)
  at com.marklogic.client.impl.ServerEvaluationCallImpl.eval(ServerEvaluationCallImpl.java:153)
  at com.marklogic.client.impl.ServerEvaluationCallImpl.evalAs(ServerEvaluationCallImpl.java:144)
  at com.marklogic.hub.impl.FlowManagerImpl.deleteDocumentsInDirectory(FlowManagerImpl.java:348)
  at com.marklogic.hub.impl.FlowManagerImpl.deleteMappingArtifacts(FlowManagerImpl.java:325)
  at com.marklogic.hub.impl.FlowManagerImpl.deleteRelatedStepArtifacts(FlowManagerImpl.java:272)
  at com.marklogic.hub.impl.FlowManagerImpl.deleteStep(FlowManagerImpl.java:199)
  at com.marklogic.hub.web.service.FlowManagerService.deleteStep(FlowManagerService.java:310)
  at com.marklogic.hub.web.web.FlowController.deleteStep(FlowController.java:155)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:566)

Reply:
Added test

Reply:
That's not human readable. Try/catch and let's clean up that error so people understand what happened and let's log it.